### ISSUE TITLE
Dec 7 - Introduction to Python for space exploration

### ISSUE THREAD

**Issue by meaghanlewis** [microsoft/ReactorPlanning#124]
# Event Information
| Category | Details |
|-----------|---------|
| Reactor Topic |  Data Science and Machine Learning | 
| Workshop Duration | 60 min | 
| Title | Introduction to Python for space exploration  |
| Description | In this workshop we explore how Python plays a role in the innovative solutions that NASA creates. Through the lens of space discovery, this workshop could ignite a passion to persistently learn, discover, and create so that you too can one day help us all understand a little more about the world beyond our Earth. We'll start with an introduction to the types of space exploration problems that Python and data science can influence, then install the tools we need for development, and write some basic Python in Visual Studio Code. If you'd like to follow along, we'll be going through this content on Microsoft Learn: https://aka.ms/PythoninSpace-2. |
| Content| https://aka.ms/SpacewithPython-8  modules 1-3 | 

# Workshop(s) details
| Date |  Local Start Time |  Location |  Speaker | SharePoint ID | Platform |
|------|-------------------|-----------|----------|---------------|----------|
| 12/8 | 8:30 PM | Bangalore | Michael Galarnyk | 11889 |----------|
| 12/8 | 4:00 PM | US, Canada, Sydney | Michael Galarnyk | 11890 |----------|
| 12/9 | 7:30 PM | Sao Paulo | Pedro Olmo Vaz de Melo | 11891 |----------|
| 12/9 | 7:30 PM | Shanghai | Ryan Chung | 11892 |----------|
| 12/9 | 12:00 PM | London, Stockholm, Abu Dhabi, Tel Aviv | Brian Sletten | 11893 |----------|

# Event Checklist

## 10 Weeks Prior
- [x] Reactor Content Team (RCT) opens a new issue for an upcoming local workshop filling in the Event Information and Workshop details tables. RCT member who owns content (content owner) this week will be the assignee.
- [x] RCT and content owner discuss details of the event in the comments of this issue. RCT adds a _confirmed-event_ label and _local-digital-event_ tag to the issue and adds event to the __Local Digital Events__ project. Issue will appear in the __Upcoming Streams__ column
- [x] RCT adds a _ready-for-marketing-edit_ label so the RCT can review title and description

## 8 Weeks Prior
- [ ] RCT does a review of title and description and removes the _ready-for-marketing-edit_ label 
- [ ] RCT moves the issue to the __Ready for SharePoint__ column
- [ ] RCT informs the Reactor Marketing Team of the new event information
- [ ] RCT confirms the speaker
- [ ] RCT provides speaker with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT moves the issue to the __Content Preparation In Progress__ column

## 5 weeks Prior
- [ ] Content owner opens a PR with the new content for the workshop on the [customer-facing Reactor repository](https://github.com/microsoft/Reactors/tree/main/online-event-resources) and adds the _content-development_ and _local-digital-event_ tag to the PR. 
- [ ] RCT reviews, approves, and merges the changes
- [ ] RCT shares content with the vendor/instructor
- [ ] RCT provides vendor/instructor with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT ensures that instructors have received an invite to the workshops and that PM team works with instructors to get setup in the platform (e.g. Teams, Teams Live Event)

## 1 Week Prior
- [ ] Connect vendor/instructor with the PM(s) for the locations where this event has been marketed
- [ ] Check-in with vendor/instructor that they are prepared for the workshop
- [ ] RCT moves the issue to the __Ready to Stream__ column

## During and After Stream
- [ ] Instructor (and optional moderator) run the stream
- [ ] _(Optional)_RCT submits recording to be uploaded to the Reactor YouTube and to be made available on LearnTV
- [ ] RCT moves issue to the __Stream Complete__ column

**Comment by mebaumb** [comment]
@meaghanlewis Let me know what you think about this:

In this workshop we explore how Python plays a role in the innovative solutions that NASA creates. Through the lens of space discovery, this workshop could ignite a passion to persistently learn, discover, and create so that you too can one day help us all understand a little more about the world beyond our Earth. We'll start with an introduction to the types of space exploration problems that Python and data science can influence, then install the tools we need for development, and write some basic Python in Visual Studio Code. If you'd like to follow along, we'll be going through this content on Microsoft Learn: https://aka.ms/PythoninSpace-2.

Pre-requisites: 
Have VSCode Downloaded <link>
Etc

**Comment by meaghanlewis** [comment]
@mebaumb this looks good! There are no prerequisites required, so we can remove that or just specify **none** under pre-reqs.

**Comment by meaghanlewis** [comment]
Here's an overview of what this workshop will cover and suggested agenda to follow with links and resources.

## Content to teach
This workshop will take Reactor community members through modules 1, 2, and 3 of the Microsoft Learn path: [Discover the role of Python in Space Exploration](https://aka.ms/SpacewithPython-8)

**SHARE THIS LINK WITH REACTOR ATTENDEES: https://aka.ms/SpacewithPython-8**

**Encourage attendees to go through the modules with you and complete it. Also encourage them to rate the module at the end.**

## Agenda
1. **Intro/Overview** 5 min
   - Welcome community members
   - Provide overview of content to be covered
   - Show the learning path
   - Encourage community members to complete and rate the modules as they go along
2. **[Introduction to Python for space exploration](https://docs.microsoft.com/en-us/learn/modules/introduction-python-nasa/)** 10-15 min
    - Talk about the opportunities of the Artemis program
    - Dive into STEM through opportunities of NASA
    - Use code and data science practices to solve space exploration problems
    - Go over the knowledge check questions, giving attendees a moment to answer on their own
3. **[Install coding tools for Python](https://docs.microsoft.com/en-us/learn/modules/install-code-tools-python-nasa/)** 10-15 min
    - Install VS Code: https://code.visualstudio.com/Download
    - Install Python: https://www.python.org/downloads/
    - Install Python extension for VS Code: https://marketplace.visualstudio.com/items?itemName=ms-python.python
4. **[Write basic Python in Notebooks in VS Code](https://docs.microsoft.com/en-us/learn/modules/basic-python-nasa/)** 20 min
    - Create a Jupyter notebook file
    - Add comments
    - Code arithmetic
    - Create variables
    - Create and manipulate text
    - Create lists of data
    - View results of variables and expressions
    - Walk through knowledge check questions
5. **Next steps** 2 min
    - Encourage attendees to complete the 2 remaining modules in the learning path so they can get a trophy, and learn more about Python through the lens of space exploration 😄 
    - Let attendees know that next week's workshop will cover [using machine learning to predict rocket launch delays](https://aka.ms/RocketDelayswithML-8) and it will build off of what was covered in this weeks workshop
6. **Closing** 5 min
    - Q&A
    - **Ask attendees to take the survey: https://aka.ms/Reactor/Survey**
      - Note: The event code is listed in this issue description and/or your Reactor contact can share that with the community members during the workshop
    - Share Resources
      - Take your first steps with Python Learn path: https://aka.ms/FirstStepsWithPython-8
    - Check if the Reactor contact has other information to share

**Comment by mebaumb** [comment]
All events published except for Sao Paulo - waiting on localization of content.

**Comment by mGalarnyk** [comment]
Hey, 

I am doing two workshops on this today. Small error in the content that is rather important. 

Python wasn't named after "Money Python" as stated in Microsoft Learn: https://docs.microsoft.com/en-us/learn/modules/install-code-tools-python-nasa/4-python
It was named after Monty Python. 

Best, 
Michael Galarnyk

**Comment by mebaumb** [comment]
Hey @mGalarnyk 

Thanks for the heads up! We'll get this updated in the Learn module, but it probably won't be updated in time for your workshops! 

Please share the correct information with the learners.

Thanks again for the heads up!

**Comment by mGalarnyk** [comment]
No problem! Just wanted to note it for future events/content. 

Did the first workshop already (Bangalore) and it was fine. Too small a detail to notice in a talk anyways.

**Comment by mebaumb** [comment]
Events have been run, Tracking details recorded
